storage/replica_command.go

```diff
@@ -2194,7 +2194,21 @@ func (r *Replica) splitTrigger(
 			// flaky without a bit of a delay.
 			r.store.stopper.RunAsyncTask(func() {
 				time.Sleep(10 * time.Millisecond)
-				// TODO(bdarnell): make sure newRng hasn't been removed
+				// Make sure that newRng hasn't been removed.
+				replica, err := r.store.GetReplica(newRng.RangeID)
+				foundReplica := false
+				if err == nil {
+					for _, rep := range split.NewDesc.Replicas {
+						if rep.ReplicaID == replica.mu.replicaID {
```

Comment:
Yeah. Added locks to it and switched `newRng` to be `replica` because we've established that they're the same at the end of the function.